tantar/src/main/java/org/opencadc/tantar/BucketValidator.java

```diff
@@ -666,4 +676,54 @@ public Artifact next() {
             }
         };
     }
+
+    /**
+     * Check the StorageMetadata for a matching ObsoleteStorageLocation and if found
+     * delete the file from storage and delete the ObsoleteStorageLocation.
+     */
+    boolean isObsoleteStorageLocation(StorageMetadata storageMetadata) {
+        if (canTakeAction()) {
```

Comment:
No info level log message that an obsolete file was deleted?

A DAO method is already an atomic action: you only use explicit transaction when there are multiple DAO method calls.

If the OSL exists this method must return true, not just if cleanup is successful. The iteration has to skip this StorageLocation if it is obsolete. I would have this method throw if you cannot determine whether the StorageLocation is obsolete or not (fail to query). So: the `obsoleteStorageLocationDAO.get` has to be outside the `canTakeAction` so the return value is correct during a dryrun.

In this case, if the `delete(storageMetadata)` fails, you don't need to fail (throw IllegalStateException); the whole point of an OSL is that a failure to delete from storage doesn't have to interrupt activity and can be left to another try later since the OSL still exists. Whether a failure to delete is a reason to throw depends on what the failure is... I would probably *not* throw here as a result of failure to delete the obsolete file.

The log message right after obsoleteStorageLocationDAO.delete should be at info level and must be specific that this is an "obsolete" -- I think what is there on line ~705 would say that, but relying on toString() probably makes it uglier and more obtuse to operators than necessary. If it says "deleted ObsoleteStorageLocation[...]" that's really only saying an inventory object they don't know about  - because it is not in the model - was deleted from the database. The info message should be more carefully constructed to say that an obsolete file was removed from the system.

Reply:
Updated the logging, removed the transactions, and the return value is based on the presence of an ObsoleteStorageLocation for the file.

Reply:
Emphasis added top previous comment. The isObsoleteStorageLocation method is still wrong -- the return value depends on whether or not this is a "dryrun". That means dryrun will tell the operator one thing and then normal mode will do something else

Reply:
Not seeing that in the code. `isObsoleteStorageLocation()`
1 - gets the OSL, which could be null if not found.
2 - if an OSL was found and it's not a dry run, delete file and OSL.
3 - if an OSL was found in 1 return true, else return false. 
The return value is based only on whether an OSL was found in 1.